lib/active_merchant/billing/gateways/openpay.rb

```diff
@@ -109,14 +114,49 @@ def create_post_for_auth_or_purchase(money, creditcard, options)
         post = {}
         post[:amount] = amount(money)
         post[:method] = 'card'
-        post[:description] = options[:description]
+        post[:description] = options[:description] || "Active Merchant Purchase"
         post[:order_id] = options[:order_id]
         post[:device_session_id] = options[:device_session_id]
         post[:currency] = (options[:currency] || currency(money)).upcase
+        post[:metadata] = options[:metadata]
+        add_customer(post, options)
         add_creditcard(post, creditcard, options)
         post
       end
 
+      def add_customer(post, options)
+          if !is_customer(options)
+            if(email = options[:email])
+              customer = {}
+              customer[:email] = email
+              if (customer_opt = options[:customer])
+                customer[:external_id] = customer_opt[:external_id] if customer_opt[:external_id]
+                customer[:name] = customer_opt[:name] if customer_opt[:name]
+                customer[:last_name] = customer_opt[:last_name] if customer_opt[:last_name]
```

Comment:
For the `name` and `last_name`, we typically use the name from the credit card rather than using part of the options hash.